### ISSUE TITLE
How to select features

### ISSUE THREAD

**Issue by ShijianXu** [lasso-net/lassonet#73]
Dear authors,

I have used the code below to show the lambda path of the trianing:

```
model = LassoNetRegressor()

path = model.path(X=train_df.drop('target', axis=1), 
          y=train_df['target'],
          X_val=val_df.drop('target', axis=1),
          y_val=val_df['target'],
          return_state_dicts=True
          )

plot_path(model, test_df.drop('target', axis=1), test_df['target'], score_function=r2_score)

y_pred = model.predict(test_df.drop('target', axis=1))
# print(y_pred.device.type)     # cuda

# For regression (use a regression-specific metric, e.g., R^2 score)
score = r2_score(test_df['target'], y_pred.cpu())
print("Score:", score)

mse = mean_squared_error(test_df['target'], y_pred.cpu())
print("MSE:", mse)
```

With this plot, I can visually pick the best lambda and other hyperparams for the training.
But I want to get the best performed model. The following code does not give me the correct feature_importances, all the output are `inf`:

```
model = LassoNetRegressor(
    lambda_seq=[400],
    random_state=True
)
model.fit(train_df.drop('target', axis=1), train_df['target'], X_val=val_df.drop('target', axis=1), y_val=val_df['target'])
y_pred = model.predict(test_df.drop('target', axis=1))

score = r2_score(test_df['target'], y_pred.cpu())
print("Score:", score)

mse = mean_squared_error(test_df['target'], y_pred.cpu())
print("MSE:", mse)

print(model.feature_importances_)

# plot pred vs gt
plt.scatter(test_df['target'], y_pred.cpu(), label='LassoNet', linewidths=.1, alpha=0.4)
```

Could you please give me some help? Thank you very much!

**Comment by louisabraham** [comment]
Hi,

Feature importances are defined as lambda values.
You can either extract the lambda values from full path training or look at the linear weights of the final model (I'm not sure they capture the notion of importance).

By the way you can store the weights when computing the path and use load() to get them. Do not retrain with just one lambda value, it will not give the same results.

**Comment by ShijianXu** [comment]
Thank you very much for such a prompt reply. I am still not very clear. Let me rephrase my question.
I want to have access to the best model, especially the corresponding $\theta$ values. I want to know which features are selected and the magnitude of their corresponding $\theta$.
